crates/turborepo-lib/src/run/mod.rs

```diff
@@ -314,4 +326,157 @@ impl<'a> Run<'a> {
 
         Ok(())
     }
+
+    #[tokio::main]
+    pub async fn get_hashes(&self) -> Result<(String, TaskHashTrackerState)> {
+        let env_at_execution_start = EnvironmentVariableMap::infer();
+
+        let package_json_path = self.base.repo_root.join_component("package.json");
+        let root_package_json =
+            PackageJson::load(&package_json_path).context("failed to read package.json")?;
+
+        let opts = self.opts()?;
+
+        let is_single_package = opts.run_opts.single_package;
+
+        let pkg_dep_graph = PackageGraph::builder(&self.base.repo_root, root_package_json.clone())
+            .with_single_package_mode(opts.run_opts.single_package)
+            .build()?;
+
+        let root_turbo_json =
+            TurboJson::load(&self.base.repo_root, &root_package_json, is_single_package)?;
+
+        let root_workspace = pkg_dep_graph
+            .workspace_info(&WorkspaceName::Root)
+            .expect("must have root workspace");
```

Comment:
Not really a comment for this PR but an infallible API for the root if this is in fact the case would be nice.